base_layer/core/src/base_node/tari_pulse_service/mod.rs

```diff
@@ -0,0 +1,190 @@
+use std::{str::FromStr, time::Duration};
+
+use futures::future;
+use hickory_client::{
+    client::{AsyncDnssecClient, ClientHandle},
+    proto::{
+        iocompat::AsyncIoTokioAsStd,
+        rr::dnssec::{public_key::Rsa, SigSigner, TrustAnchor},
+        xfer::DnsMultiplexer,
+    },
+    rr::{DNSClass, Name, RData, Record, RecordType},
+    tcp::TcpClientStream,
+};
+use log::{error, info};
+use serde::{Deserialize, Serialize};
+use tari_service_framework::{async_trait, ServiceInitializationError, ServiceInitializer, ServiceInitializerContext};
+use tari_shutdown::ShutdownSignal;
+use tari_utilities::hex::Hex;
+use tokio::{net::TcpStream as TokioTcpStream, sync::watch, time};
+
+use super::LocalNodeCommsInterface;
+use crate::base_node::comms_interface::CommsInterfaceError;
+
+const LOG_TARGET: &str = "c::bn::tari_pulse";
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct TariPulseConfig {
+    pub check_interval: Option<Duration>,
+}
+
+pub struct TariPulseService {
+    dns_name: Name,
+    config: TariPulseConfig,
+}
+
+impl TariPulseService {
+    pub async fn new() -> Result<Self, anyhow::Error> {
+        let dns_name = Name::from_str("checkpoints-nextnet.tari.com").unwrap();
+        let config = TariPulseConfig {
+            check_interval: Some(Duration::from_secs(120)),
+        };
+        Ok(Self { dns_name, config })
+    }
+
+    pub fn default_trust_anchor() -> TrustAnchor {
+        const ROOT_ANCHOR_ORIG: &[u8] = include_bytes!("20326.rsa");
+        const ROOT_ANCHOR_CURRENT: &[u8] = include_bytes!("38696.rsa");
+
+        let mut anchor = TrustAnchor::new();
+        anchor.insert_trust_anchor(&Rsa::from_public_bytes(ROOT_ANCHOR_ORIG).expect("Invalid ROOT_ANCHOR_ORIG"));
+        anchor.insert_trust_anchor(&Rsa::from_public_bytes(ROOT_ANCHOR_CURRENT).expect("Invalid ROOT_ANCHOR_CURRENT"));
+        anchor
+    }
+
+    async fn get_dns_client(&self) -> Result<AsyncDnssecClient, anyhow::Error> {
+        let timeout: Duration = Duration::from_secs(5);
+        let trust_anchor = Self::default_trust_anchor();
+
+        let (stream, handle) = TcpClientStream::<AsyncIoTokioAsStd<TokioTcpStream>>::new(([1, 1, 1, 1], 53).into());
+        let dns_muxer = DnsMultiplexer::<_, SigSigner>::with_timeout(stream, handle, timeout, None);
+        let (client, bg) = AsyncDnssecClient::builder(dns_muxer)
+            .trust_anchor(trust_anchor)
+            .build()
+            .await?;
+
+        tokio::spawn(bg);
+
+        Ok(client)
+    }
+
+    async fn run(
+        &mut self,
+        mut base_node_service: LocalNodeCommsInterface,
+        notify_passed_checkpoints: watch::Sender<bool>,
+    ) {
+        let mut interval = time::interval(self.config.check_interval.unwrap());
```

Comment:
```suggestion
        let mut interval = time::interval(self.config.check_interval);
```
Never unwrap unless in a test

Reply:
Done